### DISCUSSION TITLE
Disable non-virtual displays?

### DISCUSSION THREAD

**Discussion by Wordsmith9091** [ClassicOldSong/Apollo#147] [Ideas]
I know recent Windows updates borked the ability to have Sunshine programmatically pick the primary display. Would you consider an option to have it disable all secondary displays, possibly selectable with a checkbox in Artemis or in the audio/video settings under Apollo?

As your instructions suggest, when connecting with a client for the first time, I can set the virtual display as primary (and optionally disable any other displays), which works and is retained the next time that client connects. But when connecting with a new client, if since my client is always shown as a second (actually, third, in my case) display, it's hard to bring up the Windows display settings so that I can make the change without getting back to my physical machine. It's not a critical problem, but a mild nuisance.

**Comment by ClassicOldSong** [comment]
Each client has their own display identity with Apollo's virtual display. It is by design that you connect a new client, it'll not inherit from any other client's display configurations.

If you want to keep the same identity for whatever client opening one certain app, you can check the "Use App's own identity" in the app's settings.

**Reply by Wordsmith9091** [reply]
Thanks for such a quick reply.

I'm not looking to inherent settings from one client to another - just to disable the non-primary displays on connection, either globally or as an option per client (or even better, as a selectable option in Artemis).

It looks like that's what the "headless" option in audio/video would do, but that doesn't seem to have that effect on my machine. Or was that broken by the same change to Windows that prevents programmatic selection of the primary display?

**Reply by ClassicOldSong** [reply]
Headless mode forces use of virtual display and have adjustments that make stream work with a mathine that does not have any physical display connected(even with no dummy plugs)

Your question can be solved by enabling "Use App's own identity" and adjust config in Windows once for each app that enabled the option.

**Comment by Wordsmith9091** [comment]
I may be misunderstanding, but when I connect with "headless" selected, it's not appearing to have any effect. My physical displays are staying on.

Is it meant to disable the physical displays while streaming (which is what I'd like) or just to work if there is none physically connected?

I think I may have described something unclearly - I'm not looking to set any settings on a per-app basis or to inherent settings from one client to another. Just to always use the virtual display driver as the only display while connected.

**Reply by ClassicOldSong** [reply]
Headless mode does not disable/change any other displays, it only forces use virtual display on every connection. It can also be used to avoid GPU selection issues on hybrid GPU configurations.

There's currently no built-in way to disable all other displays. You can still use MultiMonitorTool for that but due to the latest Windows upgrade, that method might not be stable as well.

**Reply by Wordsmith9091** [reply]
Gotcha. Thanks for the clarification (and again, for the quick replies).

**Reply by ClassicOldSong** [reply]
You're welcome!
